### PR TITLE
infra: update the pre-commit hook with linters

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,18 +1,8 @@
...
-    types: [python]
-    pass_filenames: false
-
-  - id: tox_integ_tests
-    name: tox integ tests
-    stages: [push]
-    language: system
-    entry: tox -e integ-tests
-    types: [python]
-    pass_filenames: false
```
Comment (speller26) [inline .pre-commit-config.yaml:18]: Why remove these? We want the tests to pass before committing.

Comment (speller26) [reply .pre-commit-config.yaml:18]: Removing integ tests makes sense, but I think unit tests are still good to have.

**Review by speller26 — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,18 +1,8 @@
...
+  - repo: local
+    hooks:
+      - id: tox_lintrs
```
Comment (speller26) [inline .pre-commit-config.yaml:4]:
```suggestion
      - id: tox_linters
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,18 +1,8 @@
...
-    entry: tox -e integ-tests
-    types: [python]
-    pass_filenames: false
+  - repo: local
+    hooks:
+      - id: tox_linters
+        name: linters
+        entry: tox -e linters_check
+        language: system
+        types: [python]
```
Comment (speller26) [inline .pre-commit-config.yaml:8]: nit
```suggestion
        types: [python]

```

File: .git-template/hooks/pre-commit [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+if [ -f .pre-commit-config.yaml ]; then
+    echo 'pre-commit configuration detected, but `pre-commit install` was never run' 1>&2
+    exit 1
+fi
+
```
Comment (speller26) [inline .git-template/hooks/pre-commit:6]:
```suggestion
```

File: tox.ini [hunk 5]
```diff
@@ -41,6 +41,17 @@ commands =
     {[testenv:black]commands}
     {[testenv:flake8]commands}
 
+# Read only linter env
+[testenv:linters_check]
```
[the remark this answers is not in the document]
Comment (math411) [reply tox.ini:45]: Good call! I missed this as an option.  I am also planning on using this test env here in our workflows: https://github.com/aws/amazon-braket-sdk-python/blob/main/.github/workflows/check-format.yml#L29. I shall add this check in for both. 
Although, I have been eyeing up [ruff](https://github.com/astral-sh/ruff) a bit more recently to take on some of this work... out of scope for the PR.

**Pull request merged**